spec/models/asset_spec.rb

```diff
@@ -198,6 +199,79 @@
 
       asset.scanned_clean!
     end
+
+    context 'when asset is already clean' do
+      let(:state) { 'clean' }
+
+      it 'does not allow the state transition' do
+        expect { asset.scanned_clean! }
+          .to raise_error(StateMachines::InvalidTransition)
+      end
+    end
+
+    context 'when asset is already infected' do
+      let(:state) { 'infected' }
+
+      it 'does not allow the state transition' do
+        expect { asset.scanned_clean! }
+          .to raise_error(StateMachines::InvalidTransition)
+      end
+    end
+
+    context 'when asset is already uploaded' do
+      let(:state) { 'uploaded' }
+
+      # before do
+      #   asset.file = load_fixture_file('asset.png')
+      # end
+
+      it 'does not allow the state transition' do
+        expect { asset.scanned_clean! }
+          .to raise_error(StateMachines::InvalidTransition)
+      end
+    end
+  end
+
+  describe 'when an asset is marked as infected' do
+    let(:state) { 'unscanned' }
+    let(:asset) { FactoryBot.build(:asset, state: state) }
+
+    it 'does not schedule saving the asset to cloud storage' do
+      expect(SaveToCloudStorageWorker).not_to receive(:perform_async).with(asset.id)
+
+      asset.scanned_infected!
+    end
+
+    context 'when asset is clean' do
+      let(:state) { 'clean' }
+
+      it 'does not allow the state transition' do
+        expect { asset.scanned_infected! }
+          .to raise_error(StateMachines::InvalidTransition)
+      end
+    end
+
+    context 'when asset is already infected' do
+      let(:state) { 'infected' }
+
+      it 'does not allow the state transition' do
+        expect { asset.scanned_infected! }
+          .to raise_error(StateMachines::InvalidTransition)
+      end
+    end
+
+    context 'when asset is already uploaded' do
+      let(:state) { 'uploaded' }
+
+      # before do
+      #   asset.file = load_fixture_file('asset.png')
+      # end
```

Comment:
It looks as though this commented out `before` block can be removed.

Reply:
@chrisroos Well spotted - thanks!